service/worker/parentclosepolicy/workflow.go

```diff
@@ -107,52 +115,166 @@ func ProcessorWorkflow(ctx workflow.Context) error {
-				Domain: domainName,
-				WorkflowExecution: &types.WorkflowExecution{
-					WorkflowID: execution.WorkflowID,
-					RunID:      execution.RunID,
+			cancelReq := &types.HistoryRequestCancelWorkflowExecutionRequest{
+				DomainUUID: domainID,
+				CancelRequest: &types.RequestCancelWorkflowExecutionRequest{
+					Domain: domainName,
+					WorkflowExecution: &types.WorkflowExecution{
+						WorkflowID: execution.WorkflowID,
+						RunID:      execution.RunID,
+					},
+					Identity: processorWFTypeName,
 				},
-				Identity: processorWFTypeName,
-			})
+			}
+			if childWorkflowOnly {
+				cancelReq.ChildWorkflowOnly = true
+				cancelReq.ExternalWorkflowExecution = &request.ParentExecution
+			}
+			err = historyClient.RequestCancelWorkflowExecution(ctx, cancelReq)
+		default:
+			err = fmt.Errorf("unknown parent close policy: %v", execution.Policy)
 		}
-
 		if err != nil {
 			switch err.(type) {
-			case *types.EntityNotExistsError, *types.CancellationAlreadyRequestedError:
+			case *types.EntityNotExistsError,
+				*types.WorkflowExecutionAlreadyCompletedError,
+				*types.CancellationAlreadyRequestedError:
 				err = nil
+			case *types.DomainNotActiveError:
+				var domainEntry *cache.DomainCacheEntry
+				if domainEntry, err = domainCache.GetDomainByID(domainID); err == nil {
+					cluster := domainEntry.GetReplicationConfig().ActiveClusterName
+					remoteExecutions[cluster] = append(remoteExecutions[cluster], execution)
+				}
 			}
 		}
 
 		if err != nil {
-			processor.metricsClient.IncCounter(metrics.ParentClosePolicyProcessorScope, metrics.ParentClosePolicyProcessorFailures)
-			getActivityLogger(ctx).Error("failed to process parent close policy", tag.Error(err))
+			scope.IncCounter(metrics.ParentClosePolicyProcessorFailures)
+			logger.Error("Failed to process parent close policy", tag.Error(err))
 			return err
 		}
-		processor.metricsClient.IncCounter(metrics.ParentClosePolicyProcessorScope, metrics.ParentClosePolicyProcessorSuccess)
+		scope.IncCounter(metrics.ParentClosePolicyProcessorSuccess)
+	}
+
+	if err := signalRemoteCluster(
+		ctx,
+		processor.clientBean,
+		request.ParentExecution,
+		remoteExecutions,
+		processor.numWorkflows,
+	); err != nil {
+		scope.IncCounter(metrics.ParentClosePolicyProcessorFailures)
+		logger.Error("Failed to signal remote parent close policy workflow", tag.Error(err))
+		return err
 	}
+
+	return nil
+}
+
+func signalRemoteCluster(
+	ctx context.Context,
+	clientBean client.Bean,
+	parentExecution types.WorkflowExecution,
+	remoteExecutions map[string][]RequestDetail,
+	numWorkflows int,
+) error {
+	for cluster, executions := range remoteExecutions {
+		remoteClient := clientBean.GetRemoteFrontendClient(cluster)
+		signalCtx, cancel := context.WithTimeout(ctx, signalTimeout)
+		// TODO: we may want to specify a custom data converter,
+		// instead of relying on the fact that the default data converter
+		// in client library is json
+		signalInput, err := json.Marshal(Request{
```

Comment:
I think you can use the SDK's dataConverter directly?

Reply:
Good point. I didn't know the default data converter is exported in the `encoded` package. Thanks!